src/components/Input/Input.tsx

```diff
@@ -108,9 +126,15 @@ class Input extends UIComponent<any, any> {
         <ElementType {...rest} className={classes.root} {...htmlInputProps}>
           {createHTMLInput(input || type, {
             defaultProps: htmlInputProps,
```

Comment:
We can extract the creation of the input in a separate variable and reuse it in the both return statements. I would actually refactor this with the following:
```jsx
return (<ElementType {...rest} className={classes.root} {...htmlInputProps}>
          {createHTMLInput(input || type, {
            defaultProps: htmlInputProps,
            overrideProps: {
              className: inputClasses,
              ref: this.handleInputRef,
            },
          })}
          {this.computeIcon() && Icon.create(this.computeIcon(), {
            defaultProps: { className: iconClasses },
            overrideProps: predefinedProps => this.handleIconOverrides,
          })}
        </ElementType>)
```

Reply:
Thanks. Fixed.